src/main/scala/com/example/Registry/Registry.scala

```diff
@@ -0,0 +1,136 @@
+import scala.util.{Failure, Success}
+
+import pureconfig.generic.auto._
+
+final case class ScreeningTitleDB(movieTitle: String, screeningId: Int, screeningTime: Timestamp)
+final case class ScreeningTitlesDB(screeningTitles: Seq[ScreeningTitleDB])
+
+final case class ScreeningScreeningRoomDB(screeningId: Int, screeningRoomId: Int, rows: Int, seatsPerRow: Int)
+
+final case class ReservationsScreeningDB(reservations: Seq[(Int, Int)])
+
+final case class ScreeningTime(time: Timestamp)
+
+final case class ReservationCreatedDB(insertedRows: Int)
+
+object Registry {
+  sealed trait RegistryCommand
+  final case class GetScreeningTitlesDB(from: Timestamp, until: Timestamp, replyTo: ActorRef[StatusReply[ScreeningTitlesDB]]) extends RegistryCommand
+
+  final case class GetScreeningScreeningRoomDB(id: Int, replyTo: ActorRef[StatusReply[Option[ScreeningScreeningRoomDB]]]) extends RegistryCommand
+
+  final case class CreateReservationDB(reservation: Reservation, replyTo: ActorRef[StatusReply[ReservationCreatedDB]]) extends RegistryCommand
+
+  final case class GetScreeningTimeDB(id: Int, replyTo: ActorRef[StatusReply[Option[ScreeningTime]]]) extends RegistryCommand
+
+  final case class GetReservationsScreeningDB(screeningId: Int, replyTo: ActorRef[StatusReply[ReservationsScreeningDB]]) extends RegistryCommand
+
+  import slick.jdbc.H2Profile.api._
+  val db = Database.forConfig("h2mem1")
+
+  val movies = MovieSchema.movies
+  val screeningRooms = ScreeningRoomSchema.screeningRooms
+  val screenings = ScreeningSchema.screenings
+  val reservations = ReservationSchema.reservations
+
+  def apply(): Behavior[RegistryCommand] = {
+
+    implicit val timestampReader = ConfigReader[String].map(s => Timestamp.valueOf(s))
+
+    Await.result(
+      db.run {(
+          movies.schema ++
+          screeningRooms.schema ++
+          screenings.schema ++
+          reservations.schema
+        ).create
+      }, Duration(1, SECONDS))
+
+    db.run(movies ++= ConfigSource.default.loadOrThrow[MoviesDB].movies)
+    db.run(screeningRooms ++= ConfigSource.default.loadOrThrow[ScreeningRoomsDB].rooms)
+    db.run(screenings ++= ConfigSource.default.loadOrThrow[ScreeningsDB].screenings)
+
+    registry()
+  }
+
+  private def registry(): Behavior[RegistryCommand] =
+    Behaviors.receiveMessage {
+
+      case GetScreeningTitlesDB(from, until, replyTo) =>
+        db.run {
+          (for {
+            scr <- screenings                     if scr.time < until && scr.time > from
+            movie <- movies                       if movie.id === scr.movieId
+          } yield (movie.title, scr.id, scr.time))
+            .result
+        }.onComplete {
+          case Success(movies) =>                 replyTo ! StatusReply.Success(ScreeningTitlesDB(movies.map(movie => ScreeningTitleDB(movie._1, movie._2, movie._3))))
+          case Failure(exception) =>              replyTo ! StatusReply.Error(exception.getMessage)
```

Comment:
is it safe to pass the ex message to the other parts of application? what has happened with it in our code?

Reply:
why not? it is just a string in status reply error, it can be used later, for example in error response

Reply:
ok, so stacktrace should be logged for developer and simple message for end user